test/integration/good/code-gen/cpp.expected

```diff
@@ -25219,7 +25305,8 @@ s(const std::vector<T0__>& y_slice, const int& start, const int& end,
   try {
     current_statement__ = 221;
     return stan::math::reduce_sum<s_rsfunctor__>(y_slice, 1, pstream__, a, b,
-             c, d, e, f, g, h, i, j, k, l, m, n, o, p, q);
+             stan::math::eval(c), stan::math::eval(d), stan::math::eval(e),
+             f, g, h, i, j, k, l, m, n, o, p, q);
```

Comment:
This is overly cautious because the code did compile before but note that if the Stan source line were changed from
```stan
    return reduce_sum(s, y_slice, 1, a, b, c, d, e, f, g, h, i, j, k, l, m,
                      n, o, p, q);
```
to
```stan
    return reduce_sum(s, y_slice, 1, a, b, c[2:], d, e, f, g, h, i, j, k, l, m,
                      n, o, p, q);
```
it blows up without the `eval()`s.